source/MaterialXGenShader/GenOptions.h

```diff
@@ -100,7 +100,7 @@ class MX_GENSHADER_API GenOptions
     //  - graph flattening or not
 
     /// Sets the type of shader interface to be generated
-    int shaderInterfaceType;
+    ShaderInterfaceType shaderInterfaceType;
```

Comment:
Change this to be typed by the enum so that casting can be avoided and the Javascript interface calls work.